libcasr/src/js.rs

```diff
@@ -0,0 +1,390 @@
+        )))
+    }
+}
+
+/// Structure provides an interface for processing the stack trace.
+pub struct JsStacktrace;
+
+impl ParseStacktrace for JsStacktrace {
+    fn extract_stacktrace(stream: &str) -> Result<Vec<String>> {
+        // Get stack trace from JS report.
+        let re = Regex::new(r"(?m)^(?:(?:.*Error:)(?:.|\n)*?|Thrown at:\n*?)((?:\n\s*at \S+.*)+)")
+            .unwrap();
+        let Some(cap) = re.captures(stream) else {
+            return Err(Error::Casr(
+                "Couldn't find traceback in JS report".to_string(),
+            ));
+        };
+        let mut stacktrace = cap
+            .get(1)
+            .unwrap()
+            .as_str()
+            .split('\n')
+            .filter(|l| !l.is_empty())
+            .map(|l| l.trim().to_string())
+            .collect::<Vec<String>>();
+
+        // Filter out multiple spaces
+        let re = Regex::new(r" +").unwrap();
+        stacktrace = stacktrace
+            .iter()
+            .map(|e| re.replace_all(e, " ").to_string())
+            .collect();
+
+        Ok(stacktrace)
+    }
+
+    fn parse_stacktrace_entry(entry: &str) -> Result<StacktraceEntry> {
+        let mut stentry = StacktraceEntry::default();
+        let re_full = Regex::new(r"^\s*at\s+(.+?)(?:\s+(\[as.*?\])?\s*)\((.*)\)$").unwrap();
+        let re_without_pars = Regex::new(r"^\s*at\s+(.+?)(?:(?:\s+(\[as.*?\]))?\s*)$").unwrap();
+
+        /// Parse substring of `entry` related to location in some file
+        ///
+        /// # Arguments
+        ///
+        /// * `entry` - initial string for stacktrace entry
+        ///
+        /// * `stentry` - `StacktraceEntry` to save parsed location into
+        ///
+        /// * `loc` - substring to extract location information from
+        fn parse_location(entry: &str, stentry: &mut StacktraceEntry, loc: &str) -> Result<()> {
+            // filename[:line[:column]]
+            if loc.is_empty() {
+                return Err(Error::Casr(format!(
+                    "Couldn't parse location. Entry: {entry}"
+                )));
+            }
+
+            let mut debug: Vec<String> = loc.split(':').map(|s| s.to_string()).collect();
+            if debug.len() > 3 {
+                // Filename contains ':' so all the elements except
+                // the last 2 belong to filename
+                debug = loc.rsplitn(3, ':').map(|s| s.to_string()).collect();
+                debug.reverse();
+            }
+            if debug[0].contains("://") {
+                debug[0] = debug[0].rsplit("://").next().unwrap().to_string();
+            }
+            if debug[0].is_empty() || !debug[0].starts_with('/') {
+                // Filter empty filenames as related entries cannot provide enough info
+                // (e.g. "at func (:3:10)" cannot be attached to any source file)
+                // Filter non-absolute paths as they are related to internal JS modules
+                stentry.debug.file = "native".to_string();
```

Comment:
I don't think  that it is good idea, to come up with names that we couldn't find in the wild and put them into `STACK_FRAME_FILEPATH_IGNORE_REGEXES_JS` for filtering.
For this case I recommend to keep `stentry.debug.file` empty. Then while filtering we could filter entries with empty paths.